src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -435,6 +439,131 @@ struct TensorExpression<Derived, DataType, Symm, IndexList, ArgsList<Args...>>
     }
   }
 
+  /// \brief Computes the right hand side tensor multi-index that corresponds to
+  /// the left hand side tensor multi-index, according to their generic indices
```

Comment:
By multi-index I mean a concrete instantiation of the values of the indices, e.g. if we have some tensor R_{abc}, a specific component might be R_{132}, and the tensor multi-index is referring to an array of those values:  `[1, 3, 2]`. By generic indices, I'm referring to the a, b, and c in R_{abc}.

From working on this `TensorExpression` code, I've realized there are so many things we call "indices," and how easy it is to get tripped up. There are the concrete indices that refer to a component (e.g. `[1, 3, 2]`), there are the generic indices (e.g. `ti_a_t`, `ti_b_t`, `ti_c_t`), and the `TensorIndexType` indices that I still haven't found a great name for to refer to in comments (e.g. `SpatialIndex<...>` and `SpacetimeIndex<...>`). I used the phrases "tensor multi-index" and "generic indices" as described above because that's what I was seeing and hearing. If you feel there is a better or more common naming convention that is more intuitive for everyone else, please let me know, and I can make the change either in this PR or another.

(Sidenote: I really want to change all instances of `Args` and `ArgsList` to something like `TensorIndices` and `TensorIndexList` in the future, because they refer to the `TensorIndex` type and a `tmpl::list` of them, respectively. Either that, or rename them to something like `GenericIndices` and `GenericIndexList`, or whatever we want to call things like a, b, c that represent generic indices of a tensor like R_{abc})

Reply:
Thanks for clarifying! I think these are good names for the quantities of interest; it makes sense to need "new" names when getting into the details of the TEs as you are here.

I think it'd be helpful (in a future PR!) to add some documentation like what you've just written above, in some centralized piece of TE documentation. But I'm not sure where that'd be best located... What do you think of the general idea?

Reply:
I think it's a good idea and definitely plan on doing it!